### DISCUSSION TITLE
is merging similar adjacent nodes on a transformer expected?

### DISCUSSION THREAD

**Discussion by horacioh** [unifiedjs/unified#177] [Q&A] [answered]
if I have `<span>hello </span><span>world</span>`

that parsed with `rehypeParse` yields:

```js
{
  type: 'root',
  children: [
    {
      type: 'element',
      tagName: 'span',
      properties: {},
      children: [Array],
      position: [Object],
    },
    {
      type: 'element',
      tagName: 'span',
      properties: {},
      children: [Array],
      position: [Object],
    },
  ],
  data: {quirksMode: false},
  position: {
    start: {line: 1, column: 1, offset: 0},
    end: {line: 1, column: 38, offset: 37},
  },
}
```

is it Ok if in my mutation I `merge` both span nodes since they both have the same attributes? I saw that `rehype-remark` does it but I'm not sure if this is a `rehype-remark` specific behavior or something generally ecxpected foor a unifiedjs transformer.

thanks in advance!

**Comment by ChristianMurphy** [comment, marked as answer]
As long as the resulting content is still valid (in this case still a valid `<span>` tag) and the result of the transform is something you want.
Merge, split, add, and remove content to your hearts content.
